interpreter/src/ser.rs

```diff
@@ -568,8 +568,32 @@ mod tests {
     use serde_bytes::Bytes;
     use std::{collections::HashMap, iter::FromIterator, sync::Arc};
 
+    macro_rules! primitive_test {
+        ($functionName:ident, $strValue: literal, $value: expr) => {
+            #[test]
+            fn $functionName() {
+                let program = Program::compile($strValue).unwrap();
+                let result = program.execute(&Context::default());
+                assert_eq!(Value::from($value), result.unwrap());
+            }
+        };
+    }
+
+    primitive_test!(test_u64_zero, "0u", 0_u64);
+    primitive_test!(test_i64_zero, "0", 0_i64);
+    primitive_test!(test_f64_zero, "0.0", 0_f64);
+    //primitive_test!(test_f64_zero, "0.", 0_f64); this test fails
```

Comment:
Same as above ^

Reply:
This is the one number-based test which currently fails. Fixing it will require a different type of change than the other number fixes.